Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -977,7 +1017,21 @@ namespace AzToolsFramework
             }
 
             link.SetLinkDom(instance);
-
+            PrefabDom& targetTemplatePrefabDom = FindTemplateDom(targetTemplateId);
+            PrefabDomPath instancePath = link.GetInstancePath();
+            PrefabDomValue* instanceValue = instancePath.Get(targetTemplatePrefabDom);
+            if(!instanceValue)
+            {
+                PrefabDomValueReference templateInstancesReference =
+                    PrefabDomUtils::FindPrefabDomValue(targetTemplatePrefabDom, PrefabDomUtils::InstancesName);
+                PrefabDomValue& templateInstances = templateInstancesReference->get();
+                PrefabDomValue instanceObject;
+                instanceObject.SetObject();
+                templateInstances.AddMember(
+                    rapidjson::Value(instanceIterator->name.GetString(), targetTemplatePrefabDom.GetAllocator()),
+                    instanceObject.Move(), targetTemplatePrefabDom.GetAllocator());
+            }
```

Comment:
Got lost in the new code. Looks like it is trying to add instance DOM to target template DOM. Why can't it be inside `UpdateTarget()`?

Reply:
Functions should have a single responsibility to make things maintainable. The responsibility of UpdateTarget() function is to update the dom of a nested prefab within a prefab. The responsibility of ConnectTemplates() is to provide the plumbing required so that the link is correctly setup between the 2 templates. So, it makes more sense for this to be here imo. 

Also, updateTarget() is heavily used for propagation. So, it'd be an unnecessary check to add there when we know that it'll be present 99% of the time.

Reply:
I think I understand the plumbing intent now. During the first glance, I treated `if (!instanceValue)` as `if (instanceValue)` in my mind.